lib/gollum-lib/markups.rb

```diff
@@ -6,7 +6,7 @@ class Markup
         Kramdown::Document.new(content, :auto_ids => false, :input => "markdown").to_html
     }
 
-    register(:markdown,  "Markdown", :regexp => /md|mkdn?|mdown|markdown/)
+    register(:markdown,  "Markdown", :regexp => /md|mkdn?|mdown|markdown/, :h1 => /^(#\s+)(\w+.*)/)
```

Comment:
@dometto 
In case you'd be interested, I played around with the section regexes for all markups. Most of them are easy but some are downright problematic. You can see the result of my playing around in the section starting from [here](https://github.com/SkyCrawl/gollum-lib/blob/css-to-sass/lib/gollum-lib/configuration.rb#L98).